test/test.js

```diff
@@ -720,5 +721,33 @@ function tests(dbName, dbType) {
       });
     });
 
+    it('total_rows', function () {
+
+      return db.bulkDocs({docs: docs8}).then(function () {
+          var opts = {
+              fields: ['category'],
+              query: 'PL'
+            };
+          return db.search(opts);
+        }).then(function (res) {
+          res.total_rows.should.equal(3);
+        });
+    });
+
+    it('total_rows with filter and limit', function () {
+
+      return db.bulkDocs({docs: docs8}).then(function () {
+          var opts = {
+              fields: ['category'],
+              query: 'PL',
+              limit: 1,
+              filter: function (doc) { return doc.type !== "static"; }
```

Comment:
wonderful, this is very logical to me (that total_rows take filter into account, but not limit/skip) ^